src/java_bytecode/java_object_factory.cpp

```diff
@@ -181,10 +445,9 @@ Function: new_tmp_symbol
 
 \*******************************************************************/
 
-namespace {
-symbolt &new_tmp_symbol(symbol_tablet &symbol_table)
+symbolt &new_tmp_symbol(symbol_tablet &symbol_table, const std::string& prefix)
 {
-  static int temporary_counter=0;
+  static size_t temporary_counter=0;
```

Comment:
Please add a well visible TODO comment that this should possibly changed in future.